ravif/src/av1encoder.rs

```diff
@@ -43,8 +43,16 @@ pub enum BitDepth {
     #[default]
     Eight,
     Ten,
-    /// Pick 8 or 10 depending on image format and decoder compatibility
-    Auto,
```

Comment:
This now depends on what data is provided as input data, if `u16` is provided, this will by default be interpreted as 10-bit input data and when `u8` is provided it will be interpreted as 8-bit values.

Reply:
Don't get too nerdsniped into this! 

The 8-bit mode only exists as a historical curiosity caused by lack of interoperability with old Safari versions. There's no other reason to use it. 8 bits isn't precise enough for a complex codec like AV1, and 10 bits always compresses much better.

I don't want the callers to be choosing the bit depth explicitly, because they shouldn't have to know which mode they need to choose for compatibility with iOS 16.3.

The workaround for Safari is no longer needed, and I plan to delete the 8-bit encoding entirely in a few months when usage of the oldest Safari versions becomes negligible.

Reply:
Ah interesting! I was not aware of that. It seemed to me it was just a waste of storage space. Thanks for the clarification, I'll adjust the PR accordingly. For people wondering what this difference might look like I added difference images below. They are all indicative since they are displayed here as `.png` images converted from `.aviv` files since `.aviv` files can't be displayed here directly.

# Encode 8 bit source data:
![8-bit-export-Srgb](https://github.com/user-attachments/assets/df1243db-58af-499d-a38a-e14209dd14be)

# Encode 8 bit source data converted to 10 bit:
![10-bit-export-Srgb](https://github.com/user-attachments/assets/c5974a19-52af-47ee-9027-d31481d07f84)

# FLIP (perceived difference metric)
![flip 8-bit-export-Srgb 10-bit-export-Srgb 67ppd ldr](https://github.com/user-attachments/assets/058ea554-3992-4fe8-a462-a31b07808b48)